src/CommandLineParser.ts

```diff
@@ -22,9 +22,18 @@ interface CommandLineOptionOfBoolean extends CommandLineOptionBase {
     type: "boolean";
 }
 
-type CommandLineOption = CommandLineOptionOfEnum | CommandLineOptionOfBoolean;
+interface CommandLineOptionOfString extends CommandLineOptionBase {
```

Comment:
Maybe it would be better to

```ts
interface CommandLineOptionOfPrimitiveType extends CommandLineOptionBase {
    type: "string" | "boolean";
}

switch (option.type) {
    case "string":
    case "boolean": {
      if (typeof value !== option.type) {
          return {
              value: undefined,
              error: diagnosticFactories.compilerOptionRequiresAValueOfType(option.name, option.type),
          };
      }

      return { value };
    }
}
```